### ISSUE TITLE
Get the "Your system does not support Async GPU Readback" error message on my laptop with RTX 2060

### ISSUE THREAD

**Issue by averybigant** [Gameslinx/Tessellation#88]
I installed Parallex 2.0 via ckan today. The game launched successfully, but I got the "Your system does not support Async GPU Readback. (Requires DirectX 11)" error message despite the fact that I have a Geforce RTX 2060 Mobile graphics card. I am also pretty sure I am running the game on the dGPU not the iGPU (and even if it is running on iGPU, my iGPU should support DX11 as well).

It might be because I am running the Linux version of Kerbal Space Program which uses OpenGL (as far as I know), but I am not sure. I have dozens of mods installed, including EVE, Scatter and Astronomer Visual Pack. The attached are the screenshot of the error message with the detailed KSP version information at the bottom right corner, my system spec, and the list of mods I have installed.

![20220907151207_1](https://user-images.githubusercontent.com/1195580/188962437-615e8534-8fd5-4d12-a32f-539bf00f5595.jpg)
![spec](https://user-images.githubusercontent.com/1195580/188964375-3f07a046-6e58-46e4-a95b-630b9276d1ad.png)
[mods.txt](https://github.com/Gameslinx/Tessellation/files/9509180/mods.txt)

**Comment by G2-Games** [comment]
I would assume this is because the game is running with either Vulkan or OpenGL due to being on Linux

I had this same issue, the only way to get around it for me was to run it using Proton... which worked, but is a suboptimal experience. +1 for more info on this

**Comment by averybigant** [comment]
> I would assume this is because the game is running with either Vulkan or OpenGL due to being on Linux
> 
> I had this same issue, the only way to get around it for me was to run it using Proton... which worked, but is a suboptimal experience. +1 for more info on this

Did more research on this issue, [it seems Unity doesn't support async readback for opengl until 2021.2](https://forum.unity.com/threads/is-there-a-plan-to-support-asyncgpureadback-in-opengl.994630/). However, IIRC, KSP uses an earlier version of Unity. Since Parallex cannot change the Unity version of the KSP base game, I guess the only workaround is to use [this plugin](https://github.com/kidapu/AsyncGPUReadbackPlugin). I don't have any gaming or opengl programming experience, so I don't really know if it's a viable solution.

How is your experience with Proton KSP? How's the performance? and stability? Any issues / bugs you noticed?

**Comment by averybigant** [comment]
> I would assume this is because the game is running with either Vulkan or OpenGL due to being on Linux
> 
> I had this same issue, the only way to get around it for me was to run it using Proton... which worked, but is a suboptimal experience. +1 for more info on this

> I would assume this is because the game is running with either Vulkan or OpenGL due to being on Linux
> 
> I had this same issue, the only way to get around it for me was to run it using Proton... which worked, but is a suboptimal experience. +1 for more info on this

Did more research on this issue, [it seems Unity doesn't support async readback for opengl until 2021.2](https://forum.unity.com/threads/is-there-a-plan-to-support-asyncgpureadback-in-opengl.994630/). However, IIRC, KSP uses an earlier version of Unity. Since Parallex cannot change the Unity version of the KSP base game, I guess the only workaround is to use [this plugin](https://github.com/kidapu/AsyncGPUReadbackPlugin). I don't have any gaming or opengl programming experience, so I don't really know if it's a viable solution.

How is your experience with Proton KSP? How's the performance? and stability? Any issues / bugs you noticed?

Fat-fingered, reopen this mistakenly closed issue.

**Comment by averybigant** [comment]
> I would assume this is because the game is running with either Vulkan or OpenGL due to being on Linux
> 
> I had this same issue, the only way to get around it for me was to run it using Proton... which worked, but is a suboptimal experience. +1 for more info on this

Did more research on this issue, [it seems Unity doesn't support async readback for opengl until 2021.2](https://forum.unity.com/threads/is-there-a-plan-to-support-asyncgpureadback-in-opengl.994630/). However, IIRC, KSP uses an earlier version of Unity. Since Parallex cannot change the Unity version of the KSP base game, I guess the only workaround is to use [this plugin](https://github.com/kidapu/AsyncGPUReadbackPlugin). I don't have any gaming or opengl programming experience, so I don't really know if it's a viable solution.

How is your experience with Proton KSP? How's the performance? and stability? Any issues / bugs you noticed?

Fat-fingered, reopen this mistakenly closed issue.

**Comment by G2-Games** [comment]
Performance is fine for the most part normally, Parallax seems to have performance hits on some specific planets (Duna/Eeloo), but I don't know if this is due to Parallax or Proton. Otherwise KSP on proton is mostly ok, despite being not quite as good an experience as native. 

Thank you for the link to that plugin, I'll do some more looking into it myself.

**Comment by Thomas-Bastin** [comment]
I've the same issue with a 1050TI (with DX12) on a Win10.

**Comment by Thomas-Bastin** [comment]
Here My DX Diag with the same error on Win10
[DxDiag.txt](https://github.com/Gameslinx/Tessellation/files/9540043/DxDiag.txt)

**Comment by StarSmasher44** [comment]
> Here My DX Diag with the same error on Win10 [DxDiag.txt](https://github.com/Gameslinx/Tessellation/files/9540043/DxDiag.txt)

Have you tried updating your Nvidia card drivers first? by comparison to mine they are out of date.

**Comment by Gameslinx** [comment]
I will be supporting Mac/Linux using a different method to AsyncGPUReadback which has a small overhead when generating terrain quads but shouldn't be more than ~10ms per quad build which is not noticeable when added to the time it takes to build a quad (which is 100ms+ because of bad stock optimization). I will close this when support is released.

**Comment by G2-Games** [comment]
Sounds good, thank you very much!

**Comment by Gameslinx** [comment]
Update, MacOSX will never be compatible. This is because Apple dropped support for OpenGL and the latest OpenGL version for MacOSX is 4.1. Compute Shaders require OpenGL version 4.3 or later.
Sadly, this is Apple's fault. Linux and Windows people running OpenGL will still be able to use the mod, but unless you can somehow update your OpenGL version to 4.3 on MacOSX, the mod won't be available

Tl;dr linux will be supported, mac will not

**Comment by Gameslinx** [comment]
Partial support for OpenGL added
